### PR TITLE
[Narwhal] use NotifyOnce for availability of suspended certificate

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

Current `AcceptNotification` implementation is error prone and not easy to use. Switch to NotifyOnce which should achieve the intended semantics in a better way.

## Test Plan 

Verify in private testnet.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/mysten-common/src/notify_once.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ use tokio::sync::Notify;
 ///
 /// NotifyOnce guarantees that wait() will return once notify() is called,
 /// regardless of whether wait() was called before or after notify().
+#[derive(Debug)]
```
Comment (andll) [inline crates/mysten-common/src/notify_once.rs:21]: Curious why do we need derive(Debug) on it? (what useful information can we possible get from it)

Comment (mwtian) [reply crates/mysten-common/src/notify_once.rs:21]: This was to satisfy DagError, because we are wrapping NotifyOnce in DagError::Suspended.

Comment (mwtian) [reply crates/mysten-common/src/notify_once.rs:21]: We can implement a more useful Debug for whether the object is notified or not, if you prefer.

Comment (andll) [reply crates/mysten-common/src/notify_once.rs:21]: I see, I guess we can keep this as is(the right fix is probably better design for dag interface, where we don't return an debug printable Error like this, but this is clearly not related to this PR)

**Review by andll — APPROVED** [review]

**Pull request merged**